Add TOP and BOTTOM hooks on Sub Navigation

davidwhicker opened the pull request:
## Description

Wrapped current Sub Navigation UL with a DIV which then enabled the use of two new hooks to add views above or below the Sub Navigation.

## Visual changes

No visual change unless a hook is used. Please see TEST HOOK text to show the use of the BOTTOM hook. 

<img width="475" alt="Screenshot 2024-10-18 at 15 59 02" src="https://github.com/user-attachments/assets/fea1213b-314c-488b-9382-7f15c52bf783">

This is my first PR so if anything is not in the correct manner please let me know.

davidwhicker committed:
Add TOP and BOTTOM hooks on Sub Navigation

Wrapped current UL with a DIV which then enabled the use of two new hooks to add views above or below the Sub Navigation.

danharrin reviewed (changes requested):
Please rename these to `PAGE_SUB_NAVIGATION_SIDEBAR_START` and `PAGE_SUB_NAVIGATION_SIDEBAR_END`

davidwhicker committed:
name change

davidwhicker committed:
Update PanelsRenderHook.php

danharrin committed:
Update sidebar.blade.php

danharrin committed:
Update PanelsRenderHook.php

danharrin reviewed (changes requested):
I've made some changes, to make the hooks more consistent with the rest of the codebase & fix a breaking change related to the hook class. Can you check they still work please?

Please add the hooks to the documentation

davidwhicker committed:
Update 06-render-hooks.md

davidwhicker reviewed (commented):
Works for me. I have updated the docs. 

I did notice that when shrinking down to mobile size, the hook is not visible. I then tried the PAGE_SUB_NAVIGATION_SELECT_AFTER which works on mobile, however the hook is visible on Desktop ( now have both hooks visible with the same content ). I may have to dig a little deeper to get this to work nicely on mobile only as well, unless you know?

davidwhicker committed:
Mobile/Desktop view fix for PAGE_SUB_NAVIGATION_SELECT Hooks

Wrap PAGE_SUB_NAVIGATION_SELECT_BEFORE & PAGE_SUB_NAVIGATION_SELECT_AFTER with div's and apply the same class md:hidden that the select has. This stops the SELECT hooks being visible when the SELECT itself is not.

davidwhicker committed:
Update index.blade.php

danharrin committed:
Update 06-render-hooks.md

danharrin commented:
Thanks

zepfietje commented on packages/panels/resources/views/components/page/index.blade.php:
```diff
@@ -64,13 +64,17 @@
         >
             @if ($subNavigation)
-                {{ \Filament\Support\Facades\FilamentView::renderHook(\Filament\View\PanelsRenderHook::PAGE_SUB_NAVIGATION_SELECT_BEFORE, scopes: $this->getRenderHookScopes()) }}
+                <div class="md:hidden">
```
@danharrin, this breaks the UI on mobile because it adds an additional item in the flex container which has a gap:
<img width="381" alt="Screenshot 2024-11-05 at 11 17 23" src="https://github.com/user-attachments/assets/6319f8c1-38bf-4104-8fab-c52fd585941c">

danharrin replied:
Can the element be wrapped in a check to see if the render hook has output first then?

zepfietje replied:
That might be a good solution for cases where the render hook isn't used.
However, the issue would still be there when using the render hook and viewing the page on mobile I think.

danharrin replied:
If the render hook is used the space is justified IMO

zepfietje replied:
Why would it be ok then?

danharrin replied:
So there is a normal space between the content you put in the render hook and the select?

zepfietje replied:
Alright, let's apply that fix then.
